File: frontend/stores/mansei.ts
import { defineStore } from 'pinia'

interface ManseiRequest {
  name: string
  gender: 'male' | 'female'
  calendar_type: 'solar' | 'lunar'
  month_type: 'regular' | 'leap'
  birth_date: string
  birth_time?: string
}

interface ManseiResult {
  id: number
  name: string
  gender: string
  birth_info: any
  saju_pillars: {
    year: string
    month: string
    day: string
    hour: string
  }
  lunar_info: any
  five_elements: Record<string, number>
  interpretation: any
  created_at: string
}

interface ManseiState {
  currentResult: ManseiResult | null
  history: ManseiResult[]
  loading: boolean
}

export const useManseiStore = defineStore('mansei', {
  state: (): ManseiState => ({
    currentResult: null,
    history: [],
    loading: false
  }),

  getters: {
    hasResult: (state) => !!state.currentResult,
    resultCount: (state) => state.history.length
  },

  actions: {
    async calculateMansei(request: ManseiRequest) {
      const { $api } = useNuxtApp()
      const authStore = useAuthStore()
      
      this.loading = true
      
      try {
        const result = await $api('/api/v1/mansei/calculate', {
          method: 'POST',
          headers: {
            Authorization: `Bearer ${authStore.token}`
          },
          body: request
        })
        
        this.currentResult = result
        this.history.unshift(result) // 최신 결과를 앞에 추가
        
        return { success: true, result }
      } catch (error: any) {
        console.error('Mansei calculation error:', error)
        return { 
          success: false, 
          error: error.data?.detail || '만세력 계산에 실패했습니다' 
        }
      } finally {
        this.loading = false
      }
    },

    async fetchHistory() {
      const { $api } = useNuxtApp()
      const authStore = useAuthStore()
      
      try {
        const history = await $api('/api/v1/mansei/history', {
          headers: {
            Authorization: `Bearer ${authStore.token}`
          }
        })
        
        this.history = history
        return { success: true }
      } catch (error: any) {
        console.error('Fetch history error:', error)
        return { 
          success: false, 
          error: error.data?.detail || '기록 조회에 실패했습니다' 
        }
      }
    },

    async getResult(id: number) {
      const { $api } = useNuxtApp()
      const authStore = useAuthStore()
      
      try {
        const result = await $api(`/api/v1/mansei/${id}`, {
          headers: {
            Authorization: `Bearer ${authStore.token}`
          }
        })
        
        this.currentResult = result
        return { success: true, result }
      } catch (error: any) {
        console.error('Get result error:', error)
        return { 
          success: false, 
          error: error.data?.detail || '결과 조회에 실패했습니다' 
        }
      }
    },

    clearCurrentResult() {
      this.currentResult = null
    }
  }
})

File: backend/app/schemas/mansei.py
from pydantic import BaseModel, Field, validator
from typing import Optional, Dict, Any
from datetime import datetime, date, time
from enum import Enum

class Gender(str, Enum):
    MALE = "male"
    FEMALE = "female"

class CalendarType(str, Enum):
    SOLAR = "solar"
    LUNAR = "lunar"

class MonthType(str, Enum):
    REGULAR = "regular"
    LEAP = "leap"

class ManseiRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    gender: Gender
    calendar_type: CalendarType
    month_type: MonthType = MonthType.REGULAR
    birth_date: date
    birth_time: Optional[time] = None
    
    @validator('birth_date')
    def validate_birth_date(cls, v):
        if v > date.today():
            raise ValueError('생년월일은 오늘 이전이어야 합니다')
        if v.year < 1900:
            raise ValueError('1900년 이후 생년월일만 지원합니다')
        return v

class ManseiResponse(BaseModel):
    id: int
    name: str
    gender: str
    birth_info: Dict[str, Any]
    saju_pillars: Dict[str, str]
    lunar_info: Dict[str, Any]
    five_elements: Dict[str, int]
    interpretation: Dict[str, Any]
    created_at: datetime
    
    class Config:
        from_attributes = True
